api/services/CronService.js

```diff
@@ -44,6 +45,91 @@ module.exports = {
 
     });
 
-  }
+  },
 
+  splittedAggregatedDownloadstats :function(days,callback){
+    console.log('Started splitted aggregated download count');
+    ElasticSearchService.lastMonthDownloadsBulk(days,callback);
+   },
+
+  processDownloads:function(response,directDownloads,indirectDownloads,total,callback){
+    //check the response
+    if (typeof response.hits != "undefined") {    
+      var hits = response.hits.hits;
+      var promises = [];
+    }
+    else{
+      console.log("you received an undefined response, response:"+response);
+      console.log("this was probably caused because there were no stats yet for this day");
+      console.log("or processing time took over 5 minutes (the scroll interval");
+      callback();
+    }    
+    //loop over all hits
+    hits.forEach(function(hit,i) {
+        //execute queries to find inverse dependencies for all hits asynchronous, and find indirect hits before and after in ordered records
+        promises.push(Dependency.options.classMethods.findByDependantForIndependentDownloads(hit.fields.package[0]));
+        promises[promises.length-1].then(function(rootPackages){
```

Comment:
You don't need to go through `options.classMethods`, you can directly do `Dependency.findByDependantForIndependentDownloads` that's because on bootstraping, sails read all the models and construct sequelize models with it, and sequelize make all methods in `classMethods` available in the object directly